samples/snippets/csharp/how-to/strings/CompareStrings.cs

```diff
@@ -0,0 +1,255 @@
+        {
+            // <Snippet8>
+            List<string> lines = new List<string>
+            {
+                @"c:\public\textfile.txt",
+                @"c:\public\textFile.TXT",
+                @"c:\public\Text.txt",
+                @"c:\public\testfile2.txt"
+            };
+            lines.Sort((left, right) => left.CompareTo(right));
+ 
+            string searchString = @"c:\public\TEXTFILE.TXT";
+            Console.WriteLine($"Binary search for <{searchString}>");
+            int result = lines.BinarySearch(searchString, null);
+            ShowWhere<string>(lines, result);
+
+            Console.WriteLine($"{(result > 0 ? "Found" : "Did not find")} {searchString}");
+
+            void ShowWhere<T>(IList<T> collection, int index)
+            {
+                if (index < 0)
+                {
+                    index = ~index;
+
+                    Console.Write("Not found. Sorts between: ");
+
+                    if (index == 0)
+                        Console.Write("beginning of array and ");
+                    else
+                        Console.Write($"{collection[index - 1]} and ");
+
+                    if (index == collection.Count)
+                        Console.WriteLine("end of array.");
+                    else
+                        Console.WriteLine($"{collection[index]}.");
+                }
+                else
+                {
+                    Console.WriteLine($"Found at index {index}.");
+                }
+            }
+            //</snippet8>
+        }
+
+        private static void CompareAcrossCultures()
+        {
+            //<snippet9>  
+            string first = "Sie tanzen auf der Straße.";
+            string second = "Sie tanzen auf der Strasse.";
+
+            Console.WriteLine($"First sentence is <{first}>");
+            Console.WriteLine($"Second sentence is <{second}>");
+
+            System.Threading.Thread thread = System.Threading.Thread.CurrentThread;
+            System.Globalization.CultureInfo originalCulture = thread.CurrentCulture;
+
+            thread.CurrentCulture = new System.Globalization.CultureInfo("en-US");
+
+            // For culture-sensitive comparisons, use the String.Compare 
+            // overload that takes a StringComparison value.
+            int i = String.Compare(first, second, StringComparison.CurrentCulture);
+            Console.WriteLine($"Comparing in {originalCulture.EnglishName} returns {i}.");
+
+            thread.CurrentCulture = new System.Globalization.CultureInfo("de-DE");
+            i = String.Compare(first, second, StringComparison.CurrentCulture);
+            Console.WriteLine($"Comparing in {thread.CurrentCulture.Name} returns {i}.");
+
+            thread.CurrentCulture = originalCulture;
+            bool b = String.Equals(first, second, StringComparison.CurrentCulture);
+            Console.WriteLine($"The two strings {(b == true ? "are" : "are not")} equal.");
```

Comment:
Let's rename `b` to `areStringsEqual` or better name in order to replace `b == true` with just `areStringsEqual`